osu.Game/Online/Chat/MessageNotifier.cs

```diff
@@ -0,0 +1,238 @@
+                if (checkForPMs(channel, message))
+                    continue;
+
+                // change output to bool again if another "message processor" is added.
+                checkForMentions(channel, message, localUser.Value.Username);
+            }
+        }
+
+        private bool checkForPMs(Channel channel, Message message)
+        {
+            if (!notifyOnChat.Value || channel.Type != ChannelType.PM)
+                return false;
+
+            var existingNotification = privateMessageNotifications.FirstOrDefault(n => n.Username == message.Sender.Username);
+
+            if (existingNotification == null)
+            {
+                var notification = new PrivateMessageNotification(message.Sender.Username, channel, n => privateMessageNotifications.Remove(n));
+
+                notificationOverlay?.Post(notification);
+                privateMessageNotifications.Add(notification);
+            }
+            else
+            {
+                existingNotification.MessageCount++;
+            }
+
+            return true;
+        }
+
+        private void checkForMentions(Channel channel, Message message, string username)
+        {
+            if (!notifyOnMention.Value || !anyCaseInsensitive(getWords(message.Content), username))
+                return;
+
+            var notification = new MentionNotification(message.Sender.Username, channel);
+            notificationOverlay?.Post(notification);
+        }
+
+        private static IEnumerable<string> getWords(string input) => Regex.Matches(input, @"\w+", RegexOptions.Compiled).Select(c => c.Value);
+
+        /// <summary>
+        /// Finds the first matching string/word in both <paramref name="x"/> and <paramref name="y"/> (case-insensitive)
+        /// </summary>
+        private static string hasCaseInsensitive(IEnumerable<string> x, IEnumerable<string> y) => x.FirstOrDefault(x2 => anyCaseInsensitive(y, x2));
+
+        private static bool anyCaseInsensitive(IEnumerable<string> x, string y) => x.Any(x2 => x2.Equals(y, StringComparison.OrdinalIgnoreCase));
+
+        public class PrivateMessageNotification : SimpleNotification
+        {
+            public PrivateMessageNotification(string username, Channel channel, Action<PrivateMessageNotification> onRemove)
+            {
+                Icon = FontAwesome.Solid.Envelope;
+                Username = username;
+                MessageCount = 1;
+                Channel = channel;
+                OnRemove = onRemove;
+            }
+
+            private int messageCount;
+
+            public int MessageCount
+            {
+                get => messageCount;
+                set
+                {
+                    messageCount = value;
+                    Text = $"You received {"private message".ToQuantity(messageCount)} from '{Username}'. Click to read it!";
```

Comment:
number is not required.

Reply:
Why is that so?

Reply:
Due to the other comment above (don't do debounce - just shoot a notification every message for now).